google_benchmark/include/benchmark/benchmark.h

```diff
@@ -1065,6 +1065,8 @@ struct State::StateIterator {
   bool operator!=(StateIterator const&) const {
     if (BENCHMARK_BUILTIN_EXPECT(cached_ != 0, true)) return true;
 #ifdef CODSPEED_INSTRUMENTATION
+    measurement_stop();
```

Comment:
this will have to be a major release because we are essentially changing benchmark result values for end users.

I agree that this is better here, but we'll have to keep this in mind